icu4c/source/common/utrie2.cpp

```diff
@@ -574,7 +574,15 @@ enumEitherTrie(const UTrie2 *trie,
                     c+=UTRIE2_DATA_BLOCK_LENGTH;
                 } else {
                     for(j=0; j<UTRIE2_DATA_BLOCK_LENGTH; ++j) {
-                        value=enumValue(context, data32!=nullptr ? data32[block+j] : idx[block+j]);
+                        if (data32!=nullptr) {
+                            value=enumValue(context, data32[block+j]);
+                        } else if (idx!=nullptr) {
+                            value=enumValue(context, idx[block+j]);
+                        } else {
+                            /* data32 and idx are not supposed to be NULL at the same time */
+                            U_ASSERT(false);
+                            return;
+                        }
```

Comment:
No.
- A UTrie2 either stores 32-bit values (via data32), or it stores 16-bit values (via idx). One or the other is not nullptr.
- If this was ever not the case, then we would get a crash -- and we don't.
- If we did want to make this super-duper-safe, but less readable, then we would need to add real error handling here. I don't see it warranted given the invariants.